src/ctfeexpr.c

```diff
@@ -1562,7 +1560,7 @@ Expression *ctfeIndex(Loc loc, Type *type, Expression *e1, uinteger_t indx)
         StringExp *es1 = (StringExp *)e1;
         if (indx >= es1->len)
         {
-            error(loc, "string index %llu is out of bounds [0 .. %llu]", indx, (ulonglong)es1->len);
+            error(loc, "string index %llu is out of bounds [0 .. %llu]", (ulonglong)indx, (ulonglong)es1->len);
             return CTFEExp::cantexp;
```

Comment:
@ibuclaw 
For size_t related elements it is better to use %zu / %zd etc.
ie:
```
size_t x = ...;
ssize_t y = ...;
printf("%zu\n", x);  // prints as unsigned decimal
printf("%zx\n", x);  // prints as hex
printf("%zd\n", y);  // prints as signed decimal
```
These will adapt automatically to the platform dependent size of size_t.

Reply:
@ibuclaw: I did the same kind of fixes you did (using %zd in my case), just moments before you posted this PR. If you like i can attach the diff here.

Reply:
@dkgroot - Lets just stick to `%l` or `%ll`.  The explicit casts are just a formality anyway.

Reply:
Ok, it was just a suggestion.
Getting those compiler warnings silenced is the main objective